src/shogun/mathematics/Random.cpp

```diff
@@ -12,12 +12,13 @@
 #include <stdlib.h>
 #endif
 
-#include <shogun/mathematics/Random.h>
 #include <shogun/base/Parameter.h>
+#include <shogun/lib/Lock.h>
+#include <shogun/lib/Time.h>
 #include <shogun/lib/external/SFMT/SFMT.h>
 #include <shogun/lib/external/dSFMT/dSFMT.h>
-#include <shogun/lib/Time.h>
-#include <shogun/lib/Lock.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/Random.h>
```

Comment:
Hi @vigsterkr, for using global random seed in default CRandom ctor. If I write it like https://gist.github.com/MikeLing/c5d618c176a14aec06f3411ce1b30c5e I will get a **undefined symbols for architecture x86_64 c++** error. But everything works if I use CSGObject::get_global_seed() instead of sg_random_seed instead. I don't know why this happened :(